### PR TITLE
GlobalConstants support in DLScript & api.json

### PR DESCRIPTION
**Pull request by touilleMan**
Implement global constants introspection support in dlscript + referencement in api.json (see #8316)

### CODE REVIEW

File: modules/dlscript/api_generator.cpp [hunk 1]
```diff
@@ -92,6 +93,23 @@ List<ClassAPI> generate_c_api_classes() {
 	List<StringName> classes;
 	ClassDB::get_class_list(&classes);
 
+	// Register global constants as a fake GlobalConstants singleton class
+	{
+		ClassAPI global_constants_api;
+		global_constants_api.class_name = L"GlobalConstants";
+		global_constants_api.api_type = ClassDB::API_CORE;
```
Comment (touilleMan) [inline modules/dlscript/api_generator.cpp]: Not sure the value this field should have...
May it's better to keep it at `API_NONE` to give a way to determine it's not a real class.

File: modules/dlscript/godot.cpp [hunk 2]
```diff
@@ -180,6 +181,16 @@ void GDAPI *godot_dlinstance_get_userdata(godot_object *p_instance) {
 	return NULL;
 }
 
+void GDAPI godot_get_global_constants(godot_dictionary *p_constants) {
```
Comment (karroffel) [inline modules/dlscript/godot.cpp]: Why not just return a dictionary? You can use `godot_dictionary_new()` and then return that.

Comment (touilleMan) [inline modules/dlscript/godot.cpp]: My point was to make more explicit memory management should be done on user's side.
I've updated the function signature to `godot_dictionary GDAPI godot_get_global_constants()` but I'm a bit concern calling `godot_dictionary_clear(&ret_value)` to free dictionary is counter-intuitive...

**Review by karroffel — APPROVED** [review]

File: modules/dlscript/godot.cpp (hunk 2 above)
Comment (karroffel) [inline modules/dlscript/godot.cpp]: `_clear()` isn't to free. `_destroy()` is to free. Also it's not really more explicit.

```cpp
godot_dictionary d = godot_get_global_constants();

// this is the same
godot_dictionary d;
godot_get_global_constants(&d);
```

You can just assign to a more complex and managed datastructure just fine.
In fact, most C compilers implement struct return types that way anway. (reserving stack space before the call (or any other memory location for that matter), writing into the reserved memory, then return) So it's semantically almost identical to your pointer argument implementation, this is just nicer to read

**Pull request merged**